### ISSUE TITLE
Can not specify controller.nodeSelector using values.yaml

### ISSUE THREAD

**Issue by mkhon** [nginxinc/kubernetes-ingress#3683]
values-custom.yaml:

```
controller:
  nodeSelector:
    node-role.kubernetes.io/worker: "true"
```

Installation fails with the following error:
```
$ helm upgrade --install --dry-run v3 . -f values-custom.yaml
Error: UPGRADE FAILED: values don't meet the specifications of the schema(s) in the following chart(s):
nginx-ingress:
- controller.nodeSelector: nodeSelectorTerms is required
```

I am not sure the definition of nodeSelector in values.schema.json is correct. If I remove $ref there the installation succeeds.

**Comment by coolbry95** [comment]
I believe you are running into this issue #3449.

This will be fixed in 3.1.0 which is going to be release soon. #3685

**Comment by lucacome** [comment]
@mkhon this should be fixed in https://github.com/nginxinc/kubernetes-ingress/releases/tag/v3.1.0 let us know if it works for you
